src/html_support_files/odoc_search.js

```diff
@@ -0,0 +1,40 @@
+
+/* The browsers interpretation of the CORS origin policy prevents to run
+   webworkers from javascript files fetched from the file:// protocol. This hack
+   is to workaround this restriction. */
+function createWebWorker() {
+  var searchs = search_urls.map((search_url) => {
+    let parts = document.location.href.split("/");
+    parts[parts.length - 1] = search_url;
+    return '"' + parts.join("/") + '"';
+  });
+  blobContents = ['importScripts(' + searchs.join(",") + ');'];
+  var blob = new Blob(blobContents, { type: "application/javascript" });
+  var blobUrl = URL.createObjectURL(blob);
+
+  var worker = new Worker(blobUrl);
+  URL.revokeObjectURL(blobUrl);
+
+  return worker;
+}
+
+var worker = createWebWorker();
```

Comment:
This must be done lazily, for example when the search bar is first focused.
I fear that this makes documentation served over Github pages unusable as the index for Odoc itself is 28MB.

Reply:
This should be done in 9466bbec400b018bd87a5b5bfaf89ec5202762d5